packages/host/app/lib/matrix-classes/message-builder.ts

```diff
@@ -147,23 +144,51 @@ export default class MessageBuilder {
           r.event_id === this.builderContext.effectiveEventId)
       );
     }) as CommandResultEvent | undefined;
-    let status = (commandResultEvent?.content?.['m.relates_to']?.key ||
-      'ready') as CommandStatus;
-    let commandResultCardEventId =
-      commandResultEvent?.content?.msgtype ===
-      APP_BOXEL_COMMAND_RESULT_WITH_OUTPUT_MSGTYPE
-        ? commandResultEvent.content.data.cardEventId
-        : undefined;
-    let messageCommand = new MessageCommand(
+
+    return buildMessageCommand({
+      effectiveEventId: this.builderContext.effectiveEventId,
+      owner: getOwner(this)!,
       message,
-      command.id,
-      command.name,
-      command.arguments,
-      this.builderContext.effectiveEventId,
-      status,
-      commandResultCardEventId,
-      getOwner(this)!,
-    );
-    return messageCommand;
+      commandEvent: event,
+      commandResultEvent,
+    });
   }
 }
+
+export function buildMessageCommand({
+  effectiveEventId,
+  commandEvent,
+  commandResultEvent,
+  message,
+  owner,
+}: {
+  effectiveEventId: string;
+  commandEvent: CommandEvent;
+  commandResultEvent?: CommandResultEvent;
+  message: Message;
+  owner: Owner;
+}) {
+  let status = (commandResultEvent?.content['m.relates_to']?.key ||
+    'ready') as CommandStatus;
+  let commandResultCardEventId =
+    commandResultEvent?.content.msgtype ===
+    APP_BOXEL_COMMAND_RESULT_WITH_OUTPUT_MSGTYPE
+      ? commandResultEvent.content.data.cardEventId
+      : undefined;
+  let command = commandEvent.content.data.toolCall;
+  let messageCommand = new MessageCommand(
+    message,
+    command.id,
+    command.name,
+    command.arguments,
+    effectiveEventId,
+    status,
+    commandResultCardEventId,
+    owner,
+  );
+  return messageCommand;
+}
+
+export function formattedMessageForCommand(formattedBody: string) {
+  return `<p data-test-command-message class="command-message">${formattedBody}</p>`;
```

Comment:
I know this is just copied from a different position but does anyone know if we have XSS protection or the like for this? Or is it a safe assumption that malicious messages of this event type can’t be produced?

Reply:
I just added this [43de05a](https://github.com/cardstack/boxel/pull/2047/commits/43de05a15305aa015d71d4ff7278d3bc8df8cb92).